### PR TITLE
Add JAVA_HOME with some cleaning up

### PR DESCRIPTION
**Pull request by guangie88**
Add in `JAVA_HOME` env var in Dockerfile after Java installation.

Other changes:
* Update `MAINTAINER` labelling: https://docs.docker.com/v17.09/engine/reference/builder/#maintainer-deprecated
* Do `ARG` -> `ENV` pattern to allow `ENV` value to be changed in build and runtime, only for simple values: https://vsupalov.com/docker-build-time-env-values/
* Standardize all `ENV` with equal signs

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -59,15 +71,19 @@ FROM no-spark AS with-spark-optional-dag
 
 # Install Java
 RUN apt-get update \
-  && apt-get install -t jessie-backports --no-install-recommends -y openjdk-8-jre-headless \
-  && rm -rf /var/lib/apt/lists/*
+    && apt-get install -t jessie-backports --no-install-recommends -y openjdk-8-jre-headless \
+    && rm -rf /var/lib/apt/lists/*
+
+ARG java_home=/usr/lib/jvm/java-8-openjdk-amd64
+ENV JAVA_HOME=${java_home}
```
Comment (chrissng) [inline Dockerfile]: Build args should not be permitted as the installation is done in this Dockerfile.

Also it is possible to programmatically set the JAVA_HOME with the command `type -p javac | xargs readlink -f | xargs dirname | xargs dirname`

Comment (guangie88) [reply Dockerfile]: Thanks for the helpful command, will try it

Comment (guangie88) [reply Dockerfile]: As discussed, using a runtime command to set the environment var might be flakey (https://forums.docker.com/t/how-do-i-send-runs-output-to-env-in-dockerfile/16106), so will stick to setting it statically, but straight to `ENV JAVA_HOME` instead.

Comment (chrissng) [reply Dockerfile]: Yes you are indeed correct, I overlooked that this can only be executed with `RUN` not `ENV`.

**Review by chrissng — APPROVED** [review]
LGTM

**Pull request merged**